spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/Doc2Vec.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.dataset.text
+
+import java.io.FileInputStream
+import java.net.{URI, URL}
+
+import com.intel.analytics.bigdl.dataset.Transformer
+
+import scala.collection.{Iterator, immutable}
+import opennlp.tools.tokenize.{SimpleTokenizer, Tokenizer, TokenizerME, TokenizerModel}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.SparkContext
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import scala.io.Source
+
+/**
+  * Transformer that vectorize a Document (article)
+  * into a Seq[Seq[String]]
+  *
+  */
+
+class Doc2Vec(glove6BFilePath: String) extends Transformer[String, Array[Float]] {
+
+  @transient lazy val p = new Path(glove6BFilePath)
+  @transient lazy val fs = p.getFileSystem(new Configuration())
+  @transient lazy val isExist = fs.exists(p)
+
+  require(isExist, "please download glove.6B file from http://nlp.stanford.edu/data/glove.6B.zip")
+
+  val spark = SparkSession.builder().getOrCreate()
```

Comment:
I don't think you can put a spark session in the transformer. As transformer will be broadcast to workers. On workers, you can't get spark session.

Reply:
you should do this when you build the transformer. Consider moving it into the factory method of the transformer